csharp/src/Ice/ObjectAdapter.cs

```diff
@@ -1133,61 +1133,62 @@ private async Task<IReadOnlyList<Endpoint>> ComputePublishedEndpointsAsync()
             if (Communicator.TraceLevels.Transport >= 1 && endpoints.Count > 0)
             {
                 var sb = new StringBuilder("published endpoints for object adapter `");
-
-                if (Protocol == Protocol.Ice1)
-                {
-                    sb.Append(Name);
-                    sb.Append("':\n");
-                    sb.Append(string.Join(":", endpoints));
-                }
-                else
-                {
-                    sb.AppendEndpoint(endpoints[0]);
-                    if (endpoints.Count > 1)
-                    {
-                        Transport mainTransport = endpoints[0].Transport;
-                        sb.Append("?alt-endpoint=");
-                        for (int i = 1; i < endpoints.Count; ++i)
-                        {
-                            if (i > 1)
-                            {
-                                sb.Append(',');
-                            }
-                            sb.AppendEndpoint(endpoints[i], "", mainTransport != endpoints[i].Transport, '$');
-                        }
-                    }
-                }
+                sb.Append(Name);
+                sb.Append("':\n");
+                sb.AppendEndpointList(endpoints);
                 Communicator.Logger.Trace(Communicator.TraceLevels.TransportCategory, sb.ToString());
             }
 
             return endpoints;
         }
 
+        // TODO: split between register and unregister. And add cancellation token?
```

Comment:
It would indeed be good to review the ObjectAdapter async methods and add cancellation token where appropriate to pass it over this method.

Reply:
Yes. For a follow-up PR.